test/Routes/ListRoutesCommandTest.php

```diff
@@ -0,0 +1,496 @@
+            $method->invoke(
+                $this->command,
+                $this->input,
+                $this->output
+            )
+        );
+    }
+
+    /**
+     * @return array<array-key, array<array-key,string>>
+     */
+    public function sortRoutingTableDataProvider(): array
+    {
+        // phpcs:disable Generic.Files.LineLength
+        return [
+            [
+                'name',
+                '[{"name":"contact","path":"\/contact","methods":"GET","middleware":"MezzioTest\\\\Tooling\\\\Routes\\\\Middleware\\\\SimpleMiddleware"},{"name":"home","path":"\/","methods":"GET","middleware":"MezzioTest\\\\Tooling\\\\Routes\\\\Middleware\\\\ExpressMiddleware"}]',
+            ],
+            [
+                'path',
+                '[{"name":"home","path":"\/","methods":"GET","middleware":"MezzioTest\\\\Tooling\\\\Routes\\\\Middleware\\\\ExpressMiddleware"},{"name":"contact","path":"\/contact","methods":"GET","middleware":"MezzioTest\\\\Tooling\\\\Routes\\\\Middleware\\\\SimpleMiddleware"}]',
+            ],
+        ];
+        // phpcs:enable
+    }
+
+    /**
+     * @dataProvider filterRoutingTableDataProvider
+     */
+    public function testCanFilterRoutingTable(array $filterOptions, string $expectedOutput): void
+    {
+        $routes = [
+            new Route("/", new SimpleMiddleware(), ['GET'], 'home'),
+            new Route("/", new ExpressMiddleware(), ['GET'], 'home'),
+        ];
+
+        $routeCollection = $this->createMock(RouteCollector::class);
+        $routeCollection
+            ->expects($this->once())
+            ->method('getRoutes')
+            ->willReturn($routes);
+
+        /** @var ContainerInterface&MockObject $container */
+        $container = $this->createMock(ContainerInterface::class);
+        $container
+            ->expects($this->once())
+            ->method('get')
+            ->with(RouteCollector::class)
+            ->willReturn($routeCollection);
+
+        /** @var ConfigLoaderInterface $configLoader */
+        $configLoader  = $this->createMock(ConfigLoaderInterface::class);
+        $this->command = new ListRoutesCommand($container, $configLoader);
+
+        $this->input
+            ->method('getOption')
+            ->willReturnOnConsecutiveCalls(
+                'json', // format
+                false, // sort
+                false, // supports-method
+                $filterOptions['middleware'], // has-middleware
+                false, // has-name
+                false                           // has-path
+            );
+
+        $this->output
+            ->expects($this->atMost(2))
+            ->method('writeln');
+
+        $method = $this->reflectExecuteMethod();
+
+        self::assertSame(
+            0,
+            $method->invoke(
+                $this->command,
+                $this->input,
+                $this->output
+            )
+        );
+    }
+
+    /**
+     * @return array<array-key, array<array-key, mixed>>
```

Comment:
```suggestion
     * @psalm-return array<array-key, array<array<string, string>>>
```
From what I see, the JSON is not used anywhere, you can remove it - in that case my suggestion is valid - else, let me know and I can provide a different `psalm-return` that covers that string as well.